// arclight-react/dist/header/header.js
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import Styles from "./styles";
export default function Header(props) {
  const style = Styles[props.style]();
  return React.createElement("div", {
    className: style.root
  }, React.createElement(AppBar, {
    className: style.appBar,
    position: "static"
  }, React.createElement(Toolbar, null, React.createElement("img", {
    alt: "",
    src: props.data.logo.src,
    className: style.logo,
    style: {
      height: props.data.logo.height
    }
  }), React.createElement("h2", {
    className: style.title
  }, props.data.title), props.data.links.map(el => React.createElement("div", {
    onClick: () => el.cb(),
    color: "inherit"
  }, el.cnt)))));
}
