// arclight-react/dist/_SeriesSelection_/_SeriesSelection_.js
import React from "react";
import styled from "styled-components";
import { Checkbox, Img, Transition } from "../index";
import theme from "../themes";

const _SeriesSelection_ = styled.div`
  & img {
    width: 30px;
    border-radius: 50%;
  }
`;

export default /*#__PURE__*/React.memo(props => /*#__PURE__*/React.createElement(Transition, {
  trans: props.trans
}, /*#__PURE__*/React.createElement(_SeriesSelection_, null, /*#__PURE__*/React.createElement("table", {
  width: "100%"
}, /*#__PURE__*/React.createElement("tbody", null, props.data ? Object.keys(props.data).map((el, i) => /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Checkbox, {
  onCheck: () => props.cb(el)
})), props.data[el].img ? /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Img, {
  src: props.data[el].img
})) : null, /*#__PURE__*/React.createElement("td", {
  style: {
    color: theme[props.theme].lineSeries[i % theme[props.theme].lineSeries.length],
    fontWeight: "bold"
  }
}, el))) : null)))));
