// arclight-react/dist/Header/Headers/StylishCenter.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from "react";
import styled from "styled-components";
import theme from "../../themes";
const Container = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: ${props => props.width};
  height: ${props => props.height};
  background-color: transparent;
  & table {
    width: 100%;
    height: 100%;
  }
  color: ${props => theme[props.theme].textColor};
`;
const Control = styled.div`
  display: inline-block;
`;
const Crown = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;
export default class StylishCenter extends React.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "state", {
      controls1: null,
      controls2: null
    });
  }

  componentDidMount() {
    const marginMod = "6px";

    if (this.props.controls.length > 0) {
      const O = {
        c1: [],
        c2: []
      };
      this.props.controls.map((el, i) => O[i % 2 === 0 ? "c1" : "c2"].push( /*#__PURE__*/React.createElement(Control, {
        style: {
          marginRight: i % 2 === 0 ? "0px" : marginMod,
          marginLeft: i % 2 === 0 ? marginMod : "0px"
        }
      }, el)));
      this.setState({
        controls1: O.c1.reverse(),
        controls2: O.c2
      });
    }
  }

  render() {
    const marginMod = 10;
    return /*#__PURE__*/React.createElement(Container, {
      theme: this.props.theme,
      width: this.props.width,
      height: this.props.height
    }, /*#__PURE__*/React.createElement("table", null, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", {
      align: "left",
      width: "50%"
    }, this.state.controls2), /*#__PURE__*/React.createElement("td", {
      align: "right",
      width: "50%"
    }, this.state.controls1)))), /*#__PURE__*/React.createElement(Crown, null, this.props.crown), this.props.textDisplay && this.props.textDisplay.left ? /*#__PURE__*/React.createElement("div", {
      style: {
        position: "absolute",
        bottom: `${marginMod}px`,
        left: `${marginMod}px`
      }
    }, this.props.textDisplay.left) : null, this.props.textDisplay && this.props.textDisplay.center ? /*#__PURE__*/React.createElement("div", {
      style: {
        position: "absolute",
        bottom: `${marginMod}px`,
        left: "50%",
        transform: "translateX(-50%)"
      }
    }, this.props.textDisplay.center) : null, this.props.textDisplay && this.props.textDisplay.right ? /*#__PURE__*/React.createElement("div", {
      style: {
        position: "absolute",
        bottom: `${marginMod}px`,
        right: `${marginMod}px`
      }
    }, this.props.textDisplay.right) : null);
  }

}
