// arclight-react/dist/Grid/Grid.js
import React from "react";
import styled from "styled-components";
import { _SearchBar_, Transition } from "../index";
import theme from "../themes";
const Grid = styled.div`
  height: 100%;
  width: 100%;
  & table {
    width: 100%;
  }
`;
const InnerTable = styled.div`
  height: ${props => props.search ? "calc(100% - 77.5px)" : "100%"};
  width: 100%;
  overflow-y: auto;
`;
const Search = styled.div`
  margin-bottom: 20px;
`;
export default /*#__PURE__*/React.memo(props => {
  let row = [];
  let rows = [];

  const setRow = row => rows.push( /*#__PURE__*/React.createElement("tr", null, row));

  props.items.map((item, i) => {
    row.push( /*#__PURE__*/React.createElement("td", null, item));

    if ((i + 1) % props.itemsPerRow === 0) {
      setRow(row);
      row = [];
    }
  });

  if (row.length > 0) {
    setRow(row);
  }

  return /*#__PURE__*/React.createElement(Transition, {
    inheritDimensions: true,
    trans: props.trans
  }, /*#__PURE__*/React.createElement(Grid, null, props.search ? /*#__PURE__*/React.createElement(Search, null, /*#__PURE__*/React.createElement(_SearchBar_, {
    placeholder: "Search",
    filters: props.filters,
    filtersRight: props.filtersRight,
    width: "50%"
  })) : null, /*#__PURE__*/React.createElement(InnerTable, {
    theme: props.theme,
    search: props.search
  }, /*#__PURE__*/React.createElement("table", null, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, rows))))));
});
