// arclight-react/dist/alert/alert.js
import React from "react";
import Styles from "./styles";
import { Alert, AlertTitle } from "@material-ui/lab";
export default function CAlert(props) {
  const style = Styles[props.style];

  const alertType = (() => {
    switch (props.type) {
      case "success":
        return React.createElement(Alert, {
          severity: "success"
        }, props.data.text);

      case "error":
        return React.createElement(Alert, {
          severity: "error"
        }, props.data.text);

      case "warning":
        return React.createElement(Alert, {
          severity: "warning"
        }, props.data.text);

      case "info":
        return React.createElement(Alert, {
          severity: "info"
        }, props.data.text);

      case "success-title":
        return React.createElement(Alert, {
          severity: "success"
        }, React.createElement(AlertTitle, null, props.data.title), props.data.text);

      case "error-title":
        return React.createElement(Alert, {
          severity: "error"
        }, React.createElement(AlertTitle, null, props.data.title), props.data.text);

      case "warning-title":
        return React.createElement(Alert, {
          severity: "warning"
        }, React.createElement(AlertTitle, null, props.data.title), props.data.text);

      case "info-title":
        return React.createElement(Alert, {
          severity: "info"
        }, React.createElement(AlertTitle, null, props.data.title), props.data.text);
    }
  })();

  return React.createElement("div", {
    className: `alert ${props.palette}`
  }, React.createElement("div", {
    onClick: () => props.cb ? props.cb() : null,
    style: props.cb ? {
      cursor: "pointer"
    } : {}
  }, alertType));
}
