// arclight-react/dist/TextArea/TextArea.js
import React from "react";
import styled from "styled-components";
import { Button, Transition } from "../index";
import theme from "../themes";

const _TextArea = styled.div`
  width: 100%;
  border-radius: 3px;
  background-color: ${props => theme[props.props.theme].backgroundColor};
  & textarea {
    font-family: inherit;
    overflow-y: auto;
    border-radius: 3px;
    margin: 1px;
    margin-bottom: -4.5px;
    width: calc(100% - 34px);
    height: ${props => props.props.height};
    resize: none;
    padding: 15px;
    font-size: 1em;
    color: ${props => theme[theme[props.props.theme].complement].textColor};
    background-color: ${props => theme[theme[props.props.theme].complement].backgroundColor};
  }
  & textarea:focus {
    outline: none;
  }
  & textarea::selection {
    background: ${props => theme[props.props.theme].primaryColor};
  }
  & textarea::-moz-selection {
    background: ${props => theme[props.props.theme].primaryColor};
  }
`;

export default class TextArea extends React.Component {
  render() {
    return /*#__PURE__*/React.createElement(Transition, {
      trans: this.props.trans
    }, /*#__PURE__*/React.createElement(_TextArea, {
      props: this.props
    }, /*#__PURE__*/React.createElement("textarea", {
      value: this.props.value ? this.props.value : null,
      placeholder: this.props.placeholder ? this.props.placeholder : null,
      readonly: this.props.readonly,
      onChange: e => this.props.onChange ? this.props.onChange(e) : null
    })), this.props.controls ? /*#__PURE__*/React.createElement("table", {
      dir: "rtl",
      style: {
        float: "right",
        marginTop: "8px"
      }
    }, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, this.props.controls.map(el => /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Button, {
      pop: true,
      theme: theme[this.props.theme].complement,
      onClick: () => el.operation()
    }, el.name)))))) : null);
  }

}
