// arclight-react/dist/Dropdown/Dropdown.js
import React from "react";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import FormHelperText from "@material-ui/core/FormHelperText";
import FormControl from "@material-ui/core/FormControl";
import { makeStyles } from "@material-ui/core/styles";
import Select from "@material-ui/core/Select";
import styled from "styled-components";
import { Transition } from "../index";
import theme from "../themes";
const Dropdown = styled.div`
  width: 100%;
`;
const useDropDownStyles = makeStyles(() => ({
  root: {}
}));
export default /*#__PURE__*/React.memo(props => {
  const [value, setValue] = React.useState(props.items[0].value);
  return /*#__PURE__*/React.createElement(Transition, {
    trans: props.trans
  }, /*#__PURE__*/React.createElement(Dropdown, null, /*#__PURE__*/React.createElement(FormControl, {
    style: {
      width: "100%"
    },
    classes: useDropDownStyles
  }, /*#__PURE__*/React.createElement(InputLabel, null, props.id), /*#__PURE__*/React.createElement(Select, {
    value: value,
    onChange: e => {
      setValue(e.target.value);
      return props.onChange(e);
    },
    style: {
      backgroundColor: "white",
      borderRadius: "4px",
      paddingLeft: "0px",
      overflow: "hidden"
    }
  }, props.items.map((el, i) => /*#__PURE__*/React.createElement(MenuItem, {
    value: el.value,
    style: {
      color: theme[props.theme].textColor,
      backgroundColor: theme[props.theme].backgroundColor
    }
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      marginLeft: "30px"
    }
  }, el.component)))))));
});
