// arclight-react/dist/modal/modal.js
import React from "react";
import Modal from "@material-ui/core/Modal";
import Backdrop from "@material-ui/core/Backdrop";
import Fade from "@material-ui/core/Fade";
import Styles from "./styles";
export default function TransitionsModal(props) {
  const style = Styles[props.style]();
  const [open, setOpen] = React.useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return React.createElement("div", null, React.createElement("div", {
    onClick: handleOpen
  }, props.data.button), React.createElement(Modal, {
    "aria-labelledby": "transition-modal-title",
    "aria-describedby": "transition-modal-description",
    className: style.modal,
    open: open,
    onClose: handleClose,
    closeAfterTransition: true,
    BackdropComponent: Backdrop,
    BackdropProps: {
      timeout: 500
    }
  }, React.createElement(Fade, {
    in: open
  }, React.createElement("div", {
    className: style.paper
  }, React.createElement("h2", {
    id: "transition-modal-title"
  }, "Transition modal"), React.createElement("p", {
    id: "transition-modal-description"
  }, "react-transition-group animates me.")))));
}
