// arclight-react/dist/Timer/Timer.js
import React from "react";
import styled from "styled-components";
import { CountdownCircleTimer } from "react-countdown-circle-timer";
import { Transition } from "../index";
import theme from "../themes";
const Timer = styled.div``;
const size = {
  normal: 0.4,
  critical: 0.7
};
export default /*#__PURE__*/React.memo(props => /*#__PURE__*/React.createElement(Transition, {
  trans: props.trans
}, /*#__PURE__*/React.createElement(Timer, null, /*#__PURE__*/React.createElement(CountdownCircleTimer, {
  strokeLinecap: "round",
  size: props.size,
  isPlaying: props.isPlaying ? props.isPlaying : true,
  strokeWidth: props.strokeWidth,
  onComplete: () => props.onComplete ? props.onComplete() : null,
  duration: props.seconds,
  colors: theme[props.theme].timerSeries,
  trailColor: "transparent"
}, ({
  remainingTime,
  elapsedTime
}) => /*#__PURE__*/React.createElement(Transition, {
  trans: remainingTime <= props.crit ? {
    animation: "bounceIn",
    count: props.crit + 1,
    duration: 1
  } : null
}, /*#__PURE__*/React.createElement("div", {
  style: {
    fontSize: remainingTime <= props.crit ? size.critical * props.size : size.normal * props.size
  }
}, /*#__PURE__*/React.createElement(Transition, {
  trans: remainingTime <= 0 ? {
    animation: "bounceOut",
    delay: 0.5
  } : null
}, remainingTime >= 0 ? remainingTime : null)))))));
