// arclight-react/dist/Header/Header.js
import React from "react";
import styled from "styled-components";
import theme from "../themes";
import { Transition } from "../index";
import StylishCenter from "./Headers/StylishCenter";
import StylishLeft from "./Headers/StylishLeft";
const Header = styled.div``;

const router = (name, textDisplay, theme, info, controls) => {
  switch (name) {
    case "StylishCenter":
      return /*#__PURE__*/React.createElement(StylishCenter, {
        textDisplay: textDisplay,
        theme: theme,
        width: info.width,
        height: info.height,
        crown: info.crown,
        controls: controls
      });

    case "StylishLeft":
      return /*#__PURE__*/React.createElement(StylishLeft, {
        textDisplay: textDisplay,
        theme: theme,
        width: info.width,
        height: info.height,
        crown: info.crown,
        controls: controls
      });
  }
};

export default /*#__PURE__*/React.memo(props => {
  return /*#__PURE__*/React.createElement(Transition, {
    trans: props.trans
  }, /*#__PURE__*/React.createElement(Header, null, router(props.name, props.textDisplay, props.theme, props.info, props.controls)));
});
