// arclight-react/dist/_SearchBar_/_SearchBar_.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from "react";
import styled from "styled-components";
import { Transition } from "../index";
import TextField from "@material-ui/core/TextField";
import theme from "../themes";

const _SearchBar_ = styled.div`
  width: ${props => props.width};
`;

const Filter = styled.div`
  margin-left: 5px;
  margin-right: 5px;
  & img {
    width: 40px;
    height: 40px;
    border-radius: 50%;
  }
  cursor: pointer;
`;
export default class SearchBar extends React.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "state", {
      filters: this.props.filters
    });
  }

  render() {
    const search = /*#__PURE__*/React.createElement("td", {
      width: "100%",
      style: {
        minWidth: "200px"
      }
    }, /*#__PURE__*/React.createElement("form", {
      noValidate: true,
      autoComplete: "on",
      style: {
        width: "100%",
        maxWidth: "500px",
        color: "white"
      }
    }, /*#__PURE__*/React.createElement(TextField, {
      value: this.props.value ? this.props.value : null,
      label: this.props.placeholder,
      onChange: e => this.props.onChange ? this.props.onChange(e) : null,
      style: {
        width: "100%",
        color: "white"
      }
    })));

    const _filters = this.state.filters ? /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("table", null, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, this.state.filters.map((el, i) => /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement(Filter, {
      onClick: () => {
        const state = this.state;
        state.filters[i].active = !state.filters[i].active;
        el.onClick(state.filters[i]);
        this.setState(state);
      },
      style: {
        opacity: el.active ? 1 : 0.1
      }
    }, el.display))))))) : null;

    return /*#__PURE__*/React.createElement(Transition, {
      trans: this.props.trans
    }, /*#__PURE__*/React.createElement(_SearchBar_, {
      width: this.props.width
    }, /*#__PURE__*/React.createElement("table", {
      width: "100%"
    }, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, this.props.filtersRight ? /*#__PURE__*/React.createElement(React.Fragment, null, search, _filters) : /*#__PURE__*/React.createElement(React.Fragment, null, _filters, search))))));
  }

}
